src/ir/properties.h

```diff
@@ -82,8 +82,7 @@ inline bool isNamedControlFlow(Expression* curr) {
 
 inline bool isSingleConstantExpression(const Expression* curr) {
   return curr->is<Const>() || curr->is<RefNull>() || curr->is<RefFunc>() ||
-         (curr->is<I31New>() && curr->cast<I31New>()->value->is<Const>()) ||
-         curr->is<RttCanon>() || curr->is<RttSub>();
+         (curr->is<I31New>() && curr->cast<I31New>()->value->is<Const>());
```

Comment:
I added a comment now. Basically it means both that it can be used in a global initializer, and that passes that propagate constants should operate on it. I don't think they should work on `rtt.*`, or at least I'm not sure what they should do with  those values yet.

Reply:
The MVP calls out the rtt instructions as constant https://github.com/WebAssembly/gc/blob/master/proposals/gc/MVP.md#constant-expressions. Is this not implemented in our mini-MVP?

Reply:
The terms may be confusing here. `global.get` is a "constant expression" in wasm terms, that is, it can appear in a global initializer. But `isConstant` also asks about whether something should be constant propagated.

We do implement allowing RTTs to be in global initializers, that is in the MVP.

Reply:
Why can't `global.get` and `rtt.canon` be constant propagated?

Reply:
`global.get`'s value may change if it is a mutable global. (For wasm it is a "constant expression" because it can't change value in a global initializer, I guess - it has the imported or initial value, and no `global.set`s can happen before all global inits are done.)

`rtt.canon` may be constant propagated - I'm just not sure. But I'd like to leave that for future investigation after we fully understand RTT semantics.

Reply:
Then how about adding a TODO for that?

Reply:
And then can we also include `global.get` if the global is not mutable, you mean?

Reply:
There is a TODO for this in Precompute, the main relevant pass. Is that not enough do you think?

I don't think we can include `global.get` here - to find if it's mutable or not, we need to check on the module, and Precompute doesn't do a whole-module analysis atm. If it would help optimizations though then we could look into that eventually.

Reply:
Btw, note that SimplifyGlobals does do a whole-module analysis and finds constants in globals that can be propagated. That at least handles some cases, but it isn't integrated with Precompute.

Reply:
I see, didn't see the TODO. Thanks!

Reply:
I would like to have a TODO about investigating putting `RttCanon` back on this method specifically in addition to the general TODO in Precompute.